editor/editor_node.cpp

```diff
@@ -1703,9 +1701,7 @@ void EditorNode::_save_scene_with_preview(String p_file, int p_idx) {
 	save.step(TTR("Saving Scene"), 4);
 	_save_scene(p_file, p_idx);
 
-	if (!singleton->cmdline_export_mode) {
-		EditorResourcePreview::get_singleton()->check_for_invalidation(p_file);
-	}
+	EditorResourcePreview::get_singleton()->check_for_invalidation(p_file);
```

Comment:
I saw the rationale for removing the guards for `start()`, but this one doesn't seem to be a no-op on headless and will still create a MutexLock and possibly trigger more logic.

I haven't evaluated what impact this actually has, maybe it's fine. But I'm just flagging it as historically EditorResourcePreview has been the source of *many* deadlocks due to it acquiring a lock on the renderer to make previews, at the same time where we're trying to import assets also using the renderer.

Reply:
Likewise, this was also the issue with ProgressDialog and the reason for `cmdline_export_mode` to bypass both. See https://github.com/godotengine/godot/pull/85222 for details. This was a major pain during the 4.2 beta cycle, so we should tread with care here.

CC @clayjohn

Reply:
Well I guess we can give this PR a try and see if anything breaks, and if so we can go back to a more conservative approach.

Reply:
Also note that if #24031 is indeed still a problem, then that means it could potentially have been reintroduced in the context of `--import`, which is obviously live in 4.2.2 right now.